File: RepoTransBench/target_projects/C/Python/najoshi_sickle/tests/original/test_trim_single_failures.py
import subprocess
import os

def run_command(cmd):
    result = subprocess.run(cmd, shell=True, capture_output=True)
    return result

def test_trim_single_failures(monkeypatch):
    # With truncated N
    with open('test/tmp_n.fastq', 'w') as f:
        f.write("@seq1\nACGNTAAA\n+\nIIIIIIII\n")
    result = run_command("python src/main.py se -f test/tmp_n.fastq -t sanger -o out.fastq -n")
    assert result.returncode == 0

    # Quiet option
    result = run_command("python src/main.py se -f test/test.fastq -t sanger -o out.fastq --quiet")
    assert result.returncode == 0

    # No-fiveprime flag
    result = run_command("python src/main.py se -f test/test.fastq -t sanger -o out.fastq -x")
    assert result.returncode == 0

    # Clean up
    os.remove("out.fastq")
    os.remove("test/tmp_n.fastq")

File: RepoTransBench/target_projects/C/Rust/najoshi_sickle/tests/original/test_trim_single_failures.rs
use assert_cmd::Command;
use std::fs::File;
use std::io::Write;

#[test]
fn test_trim_single_failures() {
    let mut tmp_n = File::create("test/tmp_n.fastq").unwrap();
    writeln!(tmp_n, "@seq1\nACGNTAAA\n+\nIIIIIIII").unwrap();

    // sickle se with truncated N
    let mut cmd = Command::cargo_bin("najoshi_sickle").unwrap();
    cmd.args(&["se", "-f", "test/tmp_n.fastq", "-t", "sanger", "-o", "out.fastq", "-n"]);
    cmd.assert().success();

    // sickle se with quiet option
    let mut cmd = Command::cargo_bin("najoshi_sickle").unwrap();
    cmd.args(&["se", "-f", "test/test.fastq", "-t", "sanger", "-o", "out.fastq", "--quiet"]);
    cmd.assert().success();

    // sickle se with no-fiveprime
    let mut cmd = Command::cargo_bin("najoshi_sickle").unwrap();
    cmd.args(&["se", "-f", "test/test.fastq", "-t", "sanger", "-o", "out.fastq", "-x"]);
    cmd.assert().success();

    std::fs::remove_file("out.fastq").ok();
    std::fs::remove_file("test/tmp_n.fastq").ok();
}